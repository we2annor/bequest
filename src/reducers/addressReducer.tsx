import _ from "lodash";
import {
  ADD_CONTACT,
  FETCH_CONTACT,
  FETCH_CONTACTS,
  EDIT_CONTACT,
  //DELETE_CONTACT,
  AddressesState,
  AddressBookActionTypes,
} from "../actions/types";

const initialState : AddressesState = Object({})

export const addressReducer = (
  state = initialState,
  action: AddressBookActionTypes
): AddressesState => {
  switch (action.type) {
    case FETCH_CONTACT:
      return { ...state, [action.payload.id]: action.payload };
    case FETCH_CONTACTS:
      return { ...state, ..._.mapKeys(action.payload, "id") };
    case ADD_CONTACT:
      return { ...state, [action.payload.id]: action.payload };
    case EDIT_CONTACT : 
      return {...state, [action.payload.id]: action.payload};
    //case DELETE_CONTACT: 
      //return _.omit(state, action.payload);
    default:
      return state;
  }
};
