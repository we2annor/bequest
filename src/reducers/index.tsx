import {combineReducers} from 'redux';
import { addressReducer } from './addressReducer';

export const rootReducer = combineReducers({
    addressBook: addressReducer
})

export type RootState = ReturnType< typeof rootReducer>