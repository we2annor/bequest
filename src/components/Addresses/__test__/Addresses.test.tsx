import React from "react";
import { render, cleanup } from "@testing-library/react";
import { Provider } from "react-redux";
import { createStore, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import renderer from "react-test-renderer";
import { rootReducer } from "../../../reducers";
import Address from "../index";

afterEach(cleanup);

describe("When Address Mounts", () => {
  test("Renders successfully", () => {
    const store = createStore(rootReducer, applyMiddleware(thunk));
    render(
      <Provider store={store}>
        <Address />
      </Provider>
    );
  });

  test("Matches snapshot", () => {
    const store = createStore(rootReducer, applyMiddleware(thunk));
    const tree = renderer.create(
      <Provider store={store}>
        <Address />
      </Provider>
    );

    expect(tree).toMatchSnapshot();
  });
});
