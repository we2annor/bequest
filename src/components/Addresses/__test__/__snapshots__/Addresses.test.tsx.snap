// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When Address Mounts Matches snapshot 1`] = `
<div
  className="addresses"
>
  <div
    className="addresses__search-box"
    data-testid="address-search"
  >
    <input
      className="input input__field"
      onChange={[Function]}
      placeholder="Search contact"
      type="text"
    />
  </div>
  <section>
    <h2
      className="item item__heading"
    >
      Addresses
    </h2>
    <div>
      No contacts
    </div>
  </section>
</div>
`;
